presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -68,6 +72,24 @@ public HivePageSourceProvider(
     @Override
     public ConnectorPageSource createPageSource(ConnectorSession session, ConnectorSplit split, List<ColumnHandle> columns)
     {
+        if (HiveSessionProperties.getReadAsQueryUser(session)) {
+            UserGroupInformation ugi = UgiUtils.getUgi(session.getUser());
+            try {
+                return ugi.doAs((PrivilegedExceptionAction<ConnectorPageSource>) () ->
```

Comment:
I was thinking about when we make further calls to the Hive record reader (e.g., `recordReader.next`).  I would assume most readers cache the open file system input stream, but I would bet there are record readers that open new files during execution (maybe reading a sidecar file), and these would fail if the UGI is not on the thread.   Maybe we just ignore that until the problem actually happens.